src/main/java/com/google/cloud/teleport/bigtable/ParquetToBigtable.java

```diff
@@ -0,0 +1,156 @@
+ * schema. For example, if {@link BigtableCell} from the Parquet files has a 'family' of "f1", the
+ * Bigtable table should have a column family of "f1".
+ */
+public class ParquetToBigtable {
+
+  /**
+   * Options for the import pipeline.
+   */
+  public interface Options extends PipelineOptions {
+    @Description("The project that contains the table to import into.")
+    ValueProvider<String> getBigtableProjectId();
+
+    @SuppressWarnings("unused")
+    void setBigtableProjectId(ValueProvider<String> projectId);
+
+    @Description("The Bigtable instance id that contains the table to import into.")
+    ValueProvider<String> getBigtableInstanceId();
+
+    @SuppressWarnings("unused")
+    void setBigtableInstanceId(ValueProvider<String> instanceId);
+
+    @Description("The Bigtable table id to import into.")
+    ValueProvider<String> getBigtableTableId();
+
+    @SuppressWarnings("unused")
+    void setBigtableTableId(ValueProvider<String> tableId);
+
+    @Description(
+            "The input file patterm to read from. (e.g. gs://mybucket/somefolder/table1*.parquet)")
+    ValueProvider<String> getInputFilePattern();
+
+    @SuppressWarnings("unused")
+    void setInputFilePattern(ValueProvider<String> inputFilePattern);
+  }
+
+  /**
+   * Runs a pipeline to import Parquet files in GCS to a Cloud Bigtable table.
+   *
+   * @param args arguments to the pipeline
+   */
+  public static void main(String[] args) {
+    Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+
+    PipelineResult result = run(options);
+
+    // Wait for pipeline to finish only if it is not constructing a template.
+    if (options.as(DataflowPipelineOptions.class).getTemplateLocation() == null) {
+      result.waitUntilFinish();
+    }
+  }
+
+  public static PipelineResult run(Options options) {
+    Pipeline pipeline = Pipeline.create(options);
+
+    BigtableIO.Write write =
+            BigtableIO.write()
+                    .withProjectId(options.getBigtableProjectId())
+                    .withInstanceId(options.getBigtableInstanceId())
+                    .withTableId(options.getBigtableTableId());
+
+    /**
+     * Steps:
+     * 1) Read records from Parquet File.
+     * 2) Convert a GenericRecord to a KV<ByteString, Iterable<Mutation>>.
+     * 3) Write KV to Bigtable's table.
+     */
+    pipeline
+            .apply(
+                    "Read from Parquet",
+                    ParquetIO.read(BigtableRow.getClassSchema()).from(options.getInputFilePattern()))
+            .apply("Transform to Bigtable", ParDo.of(new ParquetToBigtableFn()))
+            .apply("Write to Bigtable", write);
+
+    return pipeline.run();
+  }
+
+  static class ParquetToBigtableFn
+          extends DoFn<GenericRecord, KV<ByteString, Iterable<Mutation>>> {
+
+    @ProcessElement
+    public void processElement(ProcessContext ctx) {
+      String runner = ctx.getPipelineOptions().getRunner().getCanonicalName();
```

Comment:
Instead of getting the canonical name, you can just get the Class and check whether it is assignable.

So:
Class runner = ctx.getPipelineOptions().getRunner();

And in your if statement:
if (runner.isAssignableFrom((DirectRunner.class)) {..} else if (runner.isAssignableFrom(DataflowRunner.class)) {..}

Reply:
@sabhyankar How do we handle if the Runner is not Direct or Dataflow?

Reply:
Just handle the separate condition for DirectRunner and assume that all the other runners would work similarly. So:
if (runner.isAssignableFrom((DirectRunner.class)) {..} else {..}